tests/test_TranslationDataset.py

```diff
@@ -0,0 +1,213 @@
+  """
+  Checks whether a dummy translation dataset can be read and whether the returned word indices are correct.
+  We create the necessary corpus and vocabulary files on the fly.
+  """
+
+  dummy_dataset = tempfile.mkdtemp()
+  source_file_name = os.path.join(dummy_dataset, "source.test.gz")  # testing both zipped and unzipped
+  target_file_name = os.path.join(dummy_dataset, "target.test")
+
+  with gzip.open(source_file_name, "wb") as source_file:  # writing in binary format works both in Python 2 and 3
+    source_file.write(dummy_source_text.encode("utf-8"))
+
+  with open(target_file_name, "wb") as target_file:
+    target_file.write(dummy_target_text.encode("utf-8"))
+
+  for postfix in ["", " </S>"]:  # test with and without postfix
+
+    # Replace one word by <UNK>. This way it will not appear in the vocabulary (and <UNK> is added to the vocabulary).
+    # We will test below whether this word is assigned the unknown id by checking whether the reconstruction also
+    # contains <UNK>. Note, that the input file is already written and contains the original word.
+    dummy_target_text_with_unk = dummy_target_text.replace("TranslationDatasets", "<UNK>")
+
+    # Append postfix just to have it in the vocabulary
+    source_vocabulary, inverse_source_vocabulary = create_vocabulary(dummy_source_text + postfix)
+    target_vocabulary, inverse_target_vocabulary = create_vocabulary(dummy_target_text_with_unk + postfix)
+
+    source_vocabulary_file_name = os.path.join(dummy_dataset, "source.vocab.pkl")
+    target_vocabulary_file_name = os.path.join(dummy_dataset, "target.vocab.pkl")
+
+    with open(source_vocabulary_file_name, "wb") as source_vocabulary_file:
+      pickle.dump(source_vocabulary, source_vocabulary_file)
+
+    with open(target_vocabulary_file_name, "wb") as target_vocabulary_file:
+      pickle.dump(target_vocabulary, target_vocabulary_file)
+
+    translation_dataset = TranslationDataset(
+      path=dummy_dataset, file_postfix="test", source_postfix=postfix, target_postfix=postfix,
+      unknown_label={"classes": "<UNK>"})
+    translation_dataset.init_seq_order()
+    translation_dataset.load_seqs(0, 10)
+
+    num_seqs = len(dummy_source_text.splitlines())
+    assert_equal(translation_dataset.num_seqs, num_seqs)
+
+    # Reconstruct the sentences from the word ids and compare with input.
+    for sequence_index in range(num_seqs):
+      source_word_ids = translation_dataset.get_data(sequence_index, "data")
+      source_sentence = word_ids_to_sentence(source_word_ids, inverse_source_vocabulary)
+      assert_equal(source_sentence, dummy_source_text.splitlines()[sequence_index] + postfix)
+
+      target_word_ids = translation_dataset.get_data(sequence_index, "classes")
+      target_sentence = word_ids_to_sentence(target_word_ids, inverse_target_vocabulary)
+      assert_equal(target_sentence, dummy_target_text_with_unk.splitlines()[sequence_index] + postfix)
+
+  os.system("rm -rf {}".format(dummy_dataset))
```

Comment:
Well, for a test, it does not matter too much, although it is really strange and I still would avoid that. Otherwise, no, definitely not. There are multiple issues here:
* Never use `os.system` in general. Use sth from `subprocess`.
* Just use `shutil.rmtree` here. Why do you need `/bin/rm` here? What does that give you over `shutil.rmtree`?

Reply:
Ok, will use shutil.